Polyhedron/demo/Polyhedron/Scene.cpp

```diff
@@ -1074,64 +1074,174 @@ bool Scene::dropMimeData(const QMimeData * /*data*/,
     return true;
 }
 
-void Scene::moveRowUp()
-{
-  
-  int selected_id = selectionIndices().first();
-  Scene_item* selected_item = item(selected_id);
-  if(!selected_item)
-    return;
-  if(index_map.key(selected_id).row() > 0)
+//todo : if a group is selected, don't treat it's children.
+bool Scene::sort_lists(QVector<QList<int> >&sorted_lists, bool up)
+{  
+  QVector<int> group_found;
+  Q_FOREACH(int i, selectionIndices())
   {
-    if(item(selected_id)->has_group >0)
+    Scene_item* item = this->item(i);
+    if(item->has_group == 0)
     {
-      Scene_group_item* group = selected_item->parentGroup();
-      if(group)
+      sorted_lists.first().push_back(i);
+    }
+    else
+    {
+      int group_id = item_id(item->parentGroup());
+      if(group_found.contains(group_id))
+        sorted_lists[group_id].push_back(i);
+      else
       {
-        int id = group->getChildren().indexOf(item_id(selected_item));
-        group->moveUp(id);
+        group_found.push_back(group_id);
+        if(sorted_lists.size() < group_id+1)
+          sorted_lists.resize(group_id+1);
+        sorted_lists[group_id].push_back(i);
       }
     }
-    else
+  }
+  //iterate the first list to find the groups that are selected and remove the corresponding 
+  //sub lists.
+  //todo: do that for each group. (treat subgroups)
+  for(int i = 0; i< sorted_lists.first().size(); ++i)
+  {
+    Scene_group_item* group = qobject_cast<Scene_group_item*>(this->item(sorted_lists.first()[i]));
+    if(group && ! group->getChildren().isEmpty())
+    {
+      sorted_lists[sorted_lists.first()[i]].clear();
+    }
+  }
+  std::sort(sorted_lists.first().begin(), sorted_lists.first().end(),
+            [this](int a, int b) {
+    return children.indexOf(a) < children.indexOf(b);
+});
+  if(!sorted_lists.first().isEmpty())
+  {
+    if(up &&  children.indexOf(sorted_lists.first().first()) == 0)
+      return false;
+    else if(!up &&  children.indexOf(sorted_lists.first().last()) == children.size() -1)
+      return false;
+  }
+  for(int i=1; i<sorted_lists.size(); ++i)
+  {
+    QList<int>& list = sorted_lists[i];
+    if(list.isEmpty())
+      continue;
+    Scene_group_item* group = qobject_cast<Scene_group_item*>(this->item(i));
+    if(!group)
+      continue;
+    std::sort(list.begin(), list.end(),
+              [this, group](int a, int b) {
```

Comment:
This triggers the following warning:
`warning: lambda capture 'this' is not used [-Wunused-lambda-capture]`